Lib/glyphsLib/builder.py

```diff
@@ -695,17 +695,47 @@ def add_groups_to_ufo(ufo, kerning_groups):
 
 def build_gdef(ufo):
     """Build a table GDEF statement for ligature carets."""
-    carets = {}
+    bases, ligatures, marks, carets = set(), set(), set(), {}
     for glyph in ufo:
         for anchor in glyph.anchors:
             name = anchor.get('name')
             if name and name.startswith('caret_') and 'x' in anchor:
                 carets.setdefault(glyph.name, []).append(str(anchor['x']))
-    if not carets:
+        glyphinfo = glyphsLib.glyphdata.get_glyph(glyph.name)
+        if glyphinfo:
+            category, subCategory = glyphinfo.category, glyphinfo.subCategory
+        else:
+            category, subCategory = None, None
+        # Glyphs.app assigns glyph classes like this:
+        # * Base: any glyph that has an attaching anchor and is neither
+        #   classified as Ligature nor Mark using the definitions below;
+        # * Ligature: if subCategory is "Ligature" and the glyph has at least
+        #   one attachment anchor;
+        # * Mark: if category is "Mark" and subCategory is either "Nonspacing"
+        #   or "Spacing Combining";
+        # * Compound: never assigned by Glyphs.app.
+        # https://github.com/googlei18n/glyphsLib/issues/85
+        if subCategory == 'Ligature' and glyph.anchors:
```

Comment:
by combining the two statements above, he most likely meant that the Ligature GDEF class is set if the subcategory is "Ligature" and the glyph as at least one "caret_" anchor.